src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -183,7 +183,10 @@ private ObjectNode.ObjectData EncodeEHInfo()
                 }
 
                 builder.EmitCompressedUInt((uint)clause.TryOffset);
-                builder.EmitCompressedUInt((uint)(((int)clause.TryLength << 2) | (int)clauseKind));
+
+                // clause.TryLength returned by the JIT is actually end offset...
+                int tryLength = (int)clause.TryLength - (int)clause.TryOffset;
```

Comment:
If TryLength is the end offset, I would suggest updating the variable name to reflect it accordingly. Also, does this mean RyuJIT works slightly differently for CoreCLR vs CoreRT - for the former, EHClause TryEndPC is the endOffset+1, while here it is endOffset?